### ISSUE TITLE
When to stop training MagicPoint on synthetic data?

### ISSUE THREAD

**Issue by roborocklsm** [rpautrat/SuperPoint#90]
Is it decided by precision or recall or the loss?

**Comment by rpautrat** [comment]
I recommend you to stop as soon as both the precision and recall have converged. Don't optimize MagicPoint too much because I observed overfitting when training it for too long.

**Comment by roborocklsm** [comment]
Good. Thx for your quick reply.
Could you plz give me a more clearer answer? I mean, for your experience, what are the approximate value of loss/precision/recall when it has converged. It would be a big help. I could stop it before starting over-fitting.

**Comment by rpautrat** [comment]
Sure, here is the output of one of my trainings on the synthetic dataset:
```
[10/03/2018 10:21:17 INFO] Start training
[10/03/2018 10:21:34 INFO] Iter    0: loss 4.6223, precision 0.0006, recall 0.0553
[10/03/2018 10:24:36 INFO] Iter 1000: loss 1.4267, precision 0.0028, recall 0.2565
[10/03/2018 10:26:59 INFO] Iter 2000: loss 0.4724, precision 0.0041, recall 0.3730
[10/03/2018 10:29:21 INFO] Iter 3000: loss 0.2360, precision 0.0040, recall 0.3633
[10/03/2018 10:31:41 INFO] Iter 4000: loss 0.1514, precision 0.0049, recall 0.4166
[10/03/2018 10:33:57 INFO] Iter 5000: loss 0.1196, precision 0.1377, recall 0.4705
[10/03/2018 10:36:20 INFO] Iter 6000: loss 0.0970, precision 0.1201, recall 0.3487
[10/03/2018 10:38:43 INFO] Iter 7000: loss 0.0867, precision 0.2345, recall 0.4778
[10/03/2018 10:41:06 INFO] Iter 8000: loss 0.0914, precision 0.2970, recall 0.5077
[10/03/2018 10:43:29 INFO] Iter 9000: loss 0.0847, precision 0.2614, recall 0.5091
[10/03/2018 10:45:51 INFO] Iter 10000: loss 0.0937, precision 0.2618, recall 0.4568
[10/03/2018 10:48:13 INFO] Iter 11000: loss 0.1038, precision 0.2562, recall 0.4660
[10/03/2018 10:50:36 INFO] Iter 12000: loss 0.0890, precision 0.3001, recall 0.4937
[10/03/2018 10:52:58 INFO] Iter 13000: loss 0.0799, precision 0.3186, recall 0.5071
[10/03/2018 10:55:21 INFO] Iter 14000: loss 0.1026, precision 0.2739, recall 0.4682
[10/03/2018 10:57:43 INFO] Iter 15000: loss 0.0905, precision 0.2917, recall 0.4520
[10/03/2018 11:00:06 INFO] Iter 16000: loss 0.0780, precision 0.3505, recall 0.5228
[10/03/2018 11:02:29 INFO] Iter 17000: loss 0.0760, precision 0.3343, recall 0.5054
[10/03/2018 11:04:51 INFO] Iter 18000: loss 0.0762, precision 0.3530, recall 0.5139
[10/03/2018 11:07:13 INFO] Iter 19000: loss 0.0911, precision 0.3135, recall 0.4876
[10/03/2018 11:09:35 INFO] Iter 20000: loss 0.0801, precision 0.3493, recall 0.5457
[10/03/2018 11:11:57 INFO] Iter 21000: loss 0.0952, precision 0.2958, recall 0.4951
[10/03/2018 11:14:18 INFO] Iter 22000: loss 0.0697, precision 0.3663, recall 0.5567
[10/03/2018 11:16:39 INFO] Iter 23000: loss 0.0905, precision 0.2909, recall 0.4874
[10/03/2018 11:19:00 INFO] Iter 24000: loss 0.0859, precision 0.3529, recall 0.5437
[10/03/2018 11:21:22 INFO] Iter 25000: loss 0.0672, precision 0.3019, recall 0.4773
[10/03/2018 11:23:43 INFO] Iter 26000: loss 0.0829, precision 0.3021, recall 0.4617
[10/03/2018 11:26:04 INFO] Iter 27000: loss 0.0638, precision 0.3591, recall 0.5227
[10/03/2018 11:28:25 INFO] Iter 28000: loss 0.0760, precision 0.3463, recall 0.5290
[10/03/2018 11:30:46 INFO] Iter 29000: loss 0.0916, precision 0.3030, recall 0.5364
[10/03/2018 11:33:08 INFO] Iter 30000: loss 0.0647, precision 0.3592, recall 0.5422
[10/03/2018 11:35:29 INFO] Iter 31000: loss 0.0855, precision 0.3892, recall 0.5620
[10/03/2018 11:37:50 INFO] Iter 32000: loss 0.0789, precision 0.3636, recall 0.5573
[10/03/2018 11:40:11 INFO] Iter 33000: loss 0.0715, precision 0.3241, recall 0.4999
[10/03/2018 11:42:33 INFO] Iter 34000: loss 0.0920, precision 0.4317, recall 0.5795
[10/03/2018 11:44:54 INFO] Iter 35000: loss 0.0891, precision 0.3402, recall 0.5120
[10/03/2018 11:47:15 INFO] Iter 36000: loss 0.0900, precision 0.3954, recall 0.5667
[10/03/2018 11:49:37 INFO] Iter 37000: loss 0.0753, precision 0.3764, recall 0.5662
[10/03/2018 11:51:58 INFO] Iter 38000: loss 0.0568, precision 0.2984, recall 0.4551
[10/03/2018 11:54:19 INFO] Iter 39000: loss 0.0618, precision 0.3672, recall 0.5393
[10/03/2018 11:56:40 INFO] Iter 40000: loss 0.0553, precision 0.4181, recall 0.5797
[10/03/2018 11:59:00 INFO] Training finished
[10/03/2018 11:59:01 INFO] Saving checkpoint for iteration #41000
```

**Comment by roborocklsm** [comment]
Nice! Thx.

**Comment by roborocklsm** [comment]
BTW, if the precision/recall of the network is not high enough, would it give bad performance when we use it to extract interest points from MS-COCO?

**Comment by rpautrat** [comment]
Probably, but not necessarily. It's possible to have quite low precision/recall and still a good detector. This precision and recall are mainly a proxy to monitor the progress of the network, and cannot not really be used to evaluate the performance of the network. Better to use repeatability or similar metrics to do that.
Of course if you have a good precision/recall, then your detector is probably quite good as well.

**Comment by roborocklsm** [comment]
Great. I would do the low precision version first. Thank you.
